js/components/Header.js

```diff
@@ -30,40 +32,33 @@ const Header = React.createClass ({
   componentWillReceiveProps(nextProps) {
     try {
       let title = this.state.title
-      let routeStack = nextProps.navState.routeStack
+      let path = this.state.path
+      let routeStack = nextProps.navigator.getCurrentRoutes()
       let position = routeStack.length - 1
       let nextTitle = routeStack[routeStack.length-1].title
-      if (nextTitle && nextTitle !== title) {
-        title = nextTitle
-      }
+      let nextPath = routeStack[routeStack.length-1].path
```

Comment:
I think lines 36 - 39 could be refactored. maybe by removing `position` and declare `route=routeStack[routeStack.length-1]` so we can avoid repeating it later.